frontend/src/scenes/surveys/SurveyEdit.tsx

```diff
@@ -109,6 +112,16 @@ export default function SurveyEdit(): JSX.Element {
     } = useActions(surveyLogic)
     const { surveysMultipleQuestionsAvailable } = useValues(surveysLogic)
     const { featureFlags } = useValues(enabledFeaturesLogic)
+    const sortedItemIds = survey.questions.map((_, idx) => idx.toString())
+
+    function onSortEnd({ oldIndex, newIndex }: { oldIndex: number; newIndex: number }): void {
+        function move(arr: SurveyQuestion[], from: number, to: number): SurveyQuestion[] {
+            const clone = [...arr]
+            Array.prototype.splice.call(clone, to, 0, Array.prototype.splice.call(clone, from, 1)[0])
```

Comment:
is there a reason you're not just doing `clone.splice()` ?

And is there any risk of inconsistent behaviour doing this in one statement? (since both splices are happening on the same array) - not 100% sure of evaluation order here

I'd prefer something like:

```
// Remove the element from the array
  const [element] = array.splice(fromIndex, 1);

  // Insert the element at the new position
  array.splice(toIndex, 0, element);
  ```

Reply:
I just copy pasta'd here 😅 This exists on ActionFilter for the funnel filters which I assume is more vigorously used and vetted for. I'll take a closer look though but I had just thought if it's good enough for funnel filters, it's good enough for survey questions 😆

Reply:
I think your suggestion seems good though, will add and test it out